api/class-wc-rest-dev-mailchimp-settings-controller.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ * REST API WC MailChimp settings
+ *
+ * Handles requests that interact with MailChimp plugin.
+ *
+ * @author   Automattic
+ * @category API
+ * @package  WooCommerce/API
+ * @since    3.0.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * @package WooCommerce/API
+ */
+class WC_REST_Dev_MailChimp_Settings_Controller extends WC_REST_Dev_Settings_Controller {
```

Comment:
Should this really be extending `WC_REST_Dev_Settings_Controller`? I don't see anything from it that you are using except for the permissions check. You should probably extend `WC_REST_Controller` instead and just implement your own version check (implement the function in your class).